app/components/federated_portal_button_component.rb

```diff
@@ -10,4 +11,8 @@ def initialize(name:, link:, color:, tooltip:, light_color:)
     @color = color
     @light_color = light_color
   end
+
+  def internal?
+    !link?(@link)
```

Comment:
here it should be `id.include?($REST_URL)` , because logically this function does not make sense if you are unaware of the existence of `ontoportal_ui_link` hack.

Reply:
I didn't got you well, what do you mean by `ontoportal_ui_link` hack?

Here basically, when a link is internal we pass it relative like this `/ontologies/acronym` and when it is external we pass it absolute like this `https://data...`
so the first one basically is not considered as a link this is why I used the function `link?`

Reply:
How do you enforce that ` internal we pass it relative like this /ontologies/acronym and when it is external we pass it absolute like this https://data...` ?

Reply:
For the external, no choice it should be absolute.
And for the internal onces, this is the practice that we have in all the app.
What is the case that you have in your mind and can break this?

Reply:
For the external, yes. 
For the internal, you did this 
```
 def ontoportal_ui_link(id)
    if id.include?($REST_URL)
      return id.gsub($REST_URL,'')
    end
```
which says if I understood well, check if the given link includes the internal rest URL, and in this case remove it from it. So that you enforce that the internal link is not absolute, what I propose is to not have this code and just directly test if includes or not `$REST_URL `  inside your new function  `internal?`, with no need for any external expectation of absolute or relative URL.

And also I think that this function does that already https://github.com/ontoportal-lirmm/bioportal_web_ui/blob/master/app/helpers/federation_helper.rb#L65, but to test if does work in this case.

Reply:
Discussed, ok